src/ews_serialize.erl

```diff
@@ -109,21 +126,23 @@ encode_term(Term, #elem{type=Types}=E, Tbl) when is_list(Types) ->
             Records = string:join(Aliases, ", "),
             error({"expected one of " ++ Records, Term})
     end;
-encode_term(Term, #elem{qname=Qname, type={_,_}=TypeKey}, Tbl) ->
+encode_term(Term, #elem{qname=Qname, type={_,_}=TypeKey}, Tbl) when is_tuple(Term) ->
```

Comment:
Why doesn't this clause handle `attrs`?

Reply:
That is an excellent question.

So this starts encoding of an element, and not a type, which you can see cause the type is `{_,_}` and not a `#type{}` so it is just a namespace and name at this point. It will fetch the `#type{}` and recurse in and that will be caught by a clause 3-4 down depending on the `attr` value of the `#type{}`  when it converts from the record to the all the parts needed inside this element.

```xml
<e attr="foo">
   <t1>a</t1>
   <t2>b</t2>
</e>
```

So this clause makes the `e` part of this, and the function `make_xml` actually checks the return from the next step which will make the `t`s and also return a tuple with both the attrs and the child elements.

So in a way, it does handle attrs, it is just hidden in the next call, and the `make_xml` function.

But that probably should have a comment so it is more easily spotted.